src/components/compute-type-picker.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useMemo } from 'react';
+
+import { InputLabel, MenuItem, Select, SelectChangeEvent } from '@mui/material';
+
+import { Scheduler } from '../handler';
+
+export type ComputeTypePickerProps = {
+  label: string;
+  name: string;
+  id: string;
+  environment: string;
+  onChange: (event: SelectChangeEvent<string>) => void;
+  initialValue: string;
+};
+
+export function computeTypesForEnvironment(
+  environment: string
+): string[] | null {
+  // Retrieve the environment data from session storage.
+  const environmentsData = sessionStorage.getItem('environments');
+  if (environmentsData === null) {
+    return null;
+  }
+
+  const environments = JSON.parse(
+    environmentsData
+  ) as Array<Scheduler.IRuntimeEnvironment>;
+  const environmentObj = environments.find(env => env.name === environment);
+  if (!environmentObj || !environmentObj['output_formats']) {
+    return null;
+  }
+
+  return environmentObj['compute_types'];
+}
+
+export function ComputeTypePicker(
+  props: ComputeTypePickerProps
+): JSX.Element | null {
+  const computeTypes = useMemo(
+    () => computeTypesForEnvironment(props.environment),
+    [props.environment]
+  );
```

Comment:
With the comment I made earlier, `ComputeTypePicker` can now expect a deserialized `Scheduler.IRuntimeEnvironment[]` directly as an additional prop (I suggest `envList`) Then, you can just inline the body of `computeTypesForEnvironment` within `useMemo` since it's fairly simple, and delete its original definition.

```
const computeTypes = useMemo(() => {
  const targetEnv = props.envList.find(env => env.name === props.environment);
  return targetEnv?.output_formats ? targetEnv.compute_types : null;
}, [props.envList, props.environment])
```

I would also rename the `environment` prop to `envName` to further emphasize its type as a string and to distinguish it from `envList`.

Reply:
Is `useMemo` still necessary? Now that we expect the props to be passed in and populated from the get-go, I had success removing it.

Reply:
useMemo shouldn't ever be "necessary" unless you are doing something evil like depending on only evaluating an expression on certain renders but not others. It's an optimization that basically only re-evaluates its callback whenever one of the dependencies in the second argument changes from the previous render (including the initial render). I personally prefer to use it not just as an optimization but also as a way of isolating intermediate variables (like `targetEnv`) into its own block scope.